modules/admin_manual/pages/configuration/server/security/hsmdaemon.adoc

```diff
@@ -0,0 +1,514 @@
+Install `softhsm2` with
+
+[source,console]
+----
+apt install softhsm2
+----
+
+Installing them also installs `/usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so`, which is a module that we will need later, for interaction with the PKCS11 API.
+The Debian location of the config file is `/etc/softhsm/softhsm2.conf`:
+
+[source,ini]
+....
+# SoftHSM v2 configuration file
+
+directories.tokendir = /var/lib/softhsm/tokens/
+objectstore.backend = file
+
+# ERROR, WARNING, INFO, DEBUG
+log.level = INFO
+....
+
+Make sure the `directories.tokendir` exists.
+Now we can initialize the token:
+
+[source,console]
+----
+softhsm2-util --init-token --slot 0 --label "My token 1"
+----
+
+It will ask for two PINs, a user and SO pin.
+See https://www.opendnssec.org/softhsm/, for more information.
+
+==== [optional] Installing PKCS11 CLI tools
+
+To use the PKCS11 API on the CLI, we install `opensc` with:
+
+[source,console]
+----
+apt install opensc
+----
+
+Now we can list the tokens with
+
+[source,console]
+----
+pkcs11-tool --module /usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so -l --pin 1234 -O
+----
+
+The module parameter is the library provided by the HSM vendor; in this case, it was installed with SoftHSM 2.
+See https://github.com/OpenSC/OpenSC/wiki for more information.
+
+=== Installing the hsmdaemon
+
+Installing hsmdaemon requires several steps. 
+These are:
+
+. xref:install-the-binary[Install the Binary]
+. xref:install-the-system-service[Install the System Service]
+. xref:copy-the-config-file[Copy the Config File]
+. xref:configure-the-pkcs11-module-path[Configure the PKCS 11 Module Path]
+. xref:configure-slot-and-pin[Configure Slot and Pin]
+. xref:test-key-generation[Test Key Generation]
+. xref:configure-other-options[Configure Other Options]
+. xref:owncloud-setup[ownCloud Setup]
+
+==== Install the Binary
+
+To install the hsmdaemon, you first need to install the package’s dependencies, stored in `go.mod`. To install them, run `go get`. 
+The command should not show any console output and take no more than a few minutes.
+
+Once the package dependencies are installed, you next need to compile hsmdaemon. 
+To do this, use the command: `make install`, which creates the hsmdaemon binary in the `bin` directory of your `$GOPATH`, naming it `hsmdaemon`.
+After compilation, you need to make the file executable and move it to a directory located in your system path.
+
+If you are not sure which directories are in your system path, run the following script to see a complete list:
+
+[source,console]
+----
+OFS=$IFS && IFS=':'
+for i in $(echo $PATH); do echo $i; done;
+IFS=$OFS;
+----
+
+You should see a list similar to the following:
+
+[source,console]
+----
+/usr/local/sbin
+/usr/local/bin
+/usr/sbin
+/usr/bin
+/sbin
+/bin
+----
+
+===== Installation Example
+
+To save time, copy and paste the commands below to compile the hsmdaemon.
+
+[source,console]
+----
+# Install hsmdaemon's dependencies
+go get
```

Comment:
Is `go` part of the standard distribution or does it needs to be installed first?

Reply:
Well said! I didn't think of that at the time. I'll update the content accordingly.